server/base/src/main/java/org/apache/accumulo/server/init/Initialize.java

```diff
@@ -251,16 +251,55 @@ private static boolean zookeeperAvailable(ZooReaderWriter zoo) {
     }
   }
 
+  /**
+   * Create the version directory and the instance id path and file. The method tries to create the
+   * directories and instance id file for all base directories provided unless an IOException is
+   * thrown. The IOException is not rethrown, but the method not try to create additional entries
+   * and will return false.
+   *
+   * @return false if an IOException occurred, true otherwise.
+   */
   private static boolean createDirs(VolumeManager fs, InstanceId instanceId, Set<String> baseDirs) {
+    boolean success;
+
     try {
       for (String baseDir : baseDirs) {
-        fs.mkdirs(
-            new Path(new Path(baseDir, Constants.VERSION_DIR), "" + AccumuloDataVersion.get()),
-            new FsPermission("700"));
+        log.debug("creating instance directories for base: {}", baseDir);
+
+        Path verDir =
+            new Path(new Path(baseDir, Constants.VERSION_DIR), "" + AccumuloDataVersion.get());
+        FsPermission permission = new FsPermission("700");
+
+        if (fs.exists(verDir)) {
+          FileStatus fsStat = fs.getFileStatus(verDir);
+          log.info("directory {} exists. Permissions match: {}", fsStat.getPath(),
+              fsStat.getPermission().equals(permission));
+        } else {
+          success = fs.mkdirs(verDir, permission);
+          log.info("Directory {} created - call returned {}", verDir, success);
+        }
+
         Path iidLocation = new Path(baseDir, Constants.INSTANCE_ID_DIR);
-        fs.mkdirs(iidLocation);
-        fs.createNewFile(new Path(iidLocation, instanceId.canonical()));
-        log.info("Created directory {}", baseDir);
+        if (fs.exists(iidLocation)) {
+          log.info("directory {} exists.", iidLocation);
+        } else {
+          success = fs.mkdirs(iidLocation);
+          log.info("Directory {} created - call returned {}", iidLocation, success);
+        }
+
+        Path iidPath = new Path(iidLocation, instanceId.canonical());
+
+        if (fs.exists(iidPath)) {
+          log.info("InstanceID file {} exists.", iidPath);
+        } else {
+          success = fs.createNewFile(iidPath);
+          // the exists() call provides positive check that the instanceId file is present
+          if (!success || fs.exists(iidPath)) {
+            log.info("Created instanceId file {} in hdfs", iidPath);
```

Comment:
Assuming `createNewFile()` can return false without throwing an error, I think there is still a problem with this logic. For example, if the `iidPath` doesn't exist but `fs.createNewFile(iidPath)` returns false, the `!success` will cause the if to short circuit and this message will print a misleading message.

Since we aren't quite sure of the behavior of these method calls, you could pull this chunk of logic out into a separate static method that we could be tested in a unit test.

Reply:
The logic is actually incorrect - updated it with a12c71bd27.  This PR is focused on providing log messages that we can use to debug the 'failed to find instance id' - looking at it in a broader scope, returning false instead of throwing the IOException seems like it could have been streamlined.  But, I just thought we could benefit from some additional logging and the logic could be reworked separately.